packages/ember-metal/lib/chains.js

```diff
@@ -13,8 +13,9 @@ function isObject(obj) {
   return typeof obj === 'object' && obj !== null;
 }
 
-function isVolatile(obj) {
-  return !(isObject(obj) && obj.isDescriptor && obj._volatile === false);
+function isVolatile(obj, keyName, meta) {
+  let desc = descriptorFor(obj, keyName, meta);
+  return !(desc !== undefined && desc._volatile === false);
```

Comment:
This seems a bit strange to me. I would expect `desc  !== undefined && desc._volatile !== false` (as opposed to what amounts to `desc === undefined || desc._volatile !== false` now, which I ported straight from the existing code).

In other words, this is checking "either the property is NOT a descriptor, or if it is, its `_volatile` flag is anything other than `false` (which would include `undefined` etc). I think in practice that just funnels a lot more things through `get` today, which is perhaps not a big deal?

I would like someone to check the math here. /cc @krisselden